lua/luasnip/nodes/functionNode.lua

```diff
@@ -1,7 +1,14 @@
 local FunctionNode = require("luasnip.nodes.node").Node:new()
 local util = require("luasnip.util.util")
+local lambda = require("luasnip.util.lambda")
 
 local function F(fn, args, ...)
+	if lambda.isPE(fn) then
+		local expr = lambda.instantiate(fn)
+		fn = function(args)
+			return expr(unpack(args[1]))
```

Comment:
Wouldn't the unpack 'push back' args if the snippet had more lines than one?

Reply:
If you give me an example, I can test it.

Reply:
Not snippet, srry, node. Testing should work with your snippet, just <Cr> while inside the `i(1)`.

Reply:
Yes, it works.

Reply:
In what way? Does it return two transformed lines?

Reply:
No, it keeps formating the first line only, I thought that was the expected behavior.

Reply:
Yeah, that's alright. Why not `args[1][1]` instead of `unpack(args[1])`?

Reply:
Now I `args` is a list of inputs and each input is a list of `lines`. If that's the case my implementation is incorrect as I'm passing to the function all the lines of the first input, instead of all the inputs. But managing a list of strings in a one-liner will be tricky, is fine if I replace each input with the concatenation of its lines?

Reply:
Yeah, I was going for operating on the first line only, but concatenating and splitting (return from functionNode-function should be a table of strings, each one line) after applying the function should be even more intuitive.